src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2627,14 +2627,20 @@ void InitSolarHeatGains(EnergyPlusData &state)
                                 state.dataSurface->SurfReflFacBmToDiffSolGnd)); // For linear indexing
         Real64 GndReflSolarRad = 0.0;
         Real64 GndSolarRadInc = max(state.dataEnvrn->BeamSolarRad * state.dataEnvrn->SOLCOS(3) + state.dataEnvrn->DifSolarRad, 0.0);
+
+        for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+            state.dataSurface->Surface(SurfNum).IncSolMultiplier = GetSurfIncidentSolarMultiplier(state, SurfNum);
+        }
         for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-            state.dataSurface->SurfSkySolarInc(SurfNum) = state.dataEnvrn->DifSolarRad * state.dataSolarShading->SurfAnisoSkyMult(SurfNum);
+            Real64 SurfIncSolarMultiplier = state.dataSurface->Surface(SurfNum).IncSolMultiplier;
+            state.dataSurface->SurfSkySolarInc(SurfNum) =
+                state.dataEnvrn->DifSolarRad * SurfIncSolarMultiplier * state.dataSolarShading->SurfAnisoSkyMult(SurfNum);
```

Comment:
Instead of introducing an extra multiplication for *every* surface, why not move the `GetSurfIncidentSolarMultiplier` to be after this loop and modify the final results instead, and only if there are `AnySolarMultipliers`, then this function could be renamed `ModifySurfIncidentSolar` and just apply the corrections as needed. I'm not sure if this can be done everywhere this is being applied, but we want to limit as much extra computation as possible for the common case which has no multipliers.

Reply:
It is written this way so that it is easier for me to keep track: whenever a surface has something computed using state.dataEnvrn->DifSolarRad, state.dataEnvrn->BeamSolarRad, or state.dataEnvrn->GndSolarRad, the multiplier is applied when the surface has an incident solar multiplier defined. Now that hopefully I've captured all the places where the multiplier needs to be added. I can certainly make the modification and move the solar adjustment to after the loop. I'm not sure if directly modifying `GetSurfIncidentSolarMultiplier` would work, as different variables are adjusted in different places the multiplier is used. But at least I can do something like this. What do you think? @mjwitte 

```
if (state.dataGlobal->AnySolarMultipliers) {
  for (int SurfNum : state.dataSurface->AllExtSolWindowSurfaceList) {
    Real64 SurfIncSolarMultiplier = state.dataSurface->Surface(SurfNum).IncSolMultiplier;
    variableToAdjust *= SurfIncSolarMultiplier;
  }
}

Reply:
@yujiex Yes, that's what I was suggesting. I'll leave it to you to evaluate the best approach.

Reply:
A similar approach is used for things like TDD diffuser and dome in other parts of the code, if I remember correctly.

Reply:
Thanks for the pointer. I'll check out the TDD diffuser and dome code.

Reply:
Not sure the local variable helps here since it's used only once. Just use a 1-liner.

    variableToAdjust *= state.dataSurface->Surface(SurfNum).IncSolMultiplier;

Reply:
There should be a few more terms to adjust (didn't list them all here), so I assume with a temp variable saves a little time in retrieving the multiplier value